push_notifications/models.py

```diff
@@ -206,3 +209,44 @@ def send_message(self, message, **kwargs):
 		return wns_send_message(
 			uri=self.registration_id, message=message, application_id=self.application_id, **kwargs
 		)
+
+
+class WebPushDeviceManager(models.Manager):
+	def get_queryset(self):
+		return WebPushDeviceQuerySet(self.model)
+
+
+class WebPushDeviceQuerySet(models.query.QuerySet):
+	def send_message(self, message, **kwargs):
+		devices = self.filter(active=True).order_by("application_id").distinct()
+		res = []
+		for device in devices:
+			res.append(device.send_message(message))
+		return res
+
+
+class WebPushDevice(Device):
+	registration_id = models.TextField(verbose_name=_("Registration ID"))
+	p256dh = models.TextField(verbose_name=_("User public encryption key"))
+	auth = models.TextField(verbose_name=_("User auth secret"))
+	browser = models.CharField(
```

Comment:
I'm not sure I like this model but I don't know enough there to reject it outright.

Reply:
Improve it: [41e1fd3377cc5407d46c802797d3fadd1af861c9]